### DISCUSSION TITLE
Error: Credential ID was not base64url-encoded simple webauthn

### DISCUSSION THREAD

**Discussion by nomad-pixel** [MasterKale/SimpleWebAuthn#99] [Troubleshooting]
![error](https://user-images.githubusercontent.com/75156836/107802275-517d9b00-6d8b-11eb-9141-8f23af618e07.jpg)

Credential ID was not base64url-encoded s
![erroaqr](https://user-images.githubusercontent.com/75156836/107820491-138d7080-6da5-11eb-880e-895b0a056374.jpg)
simple webauthn
![image](https://user-images.githubusercontent.com/75156836/107821279-5a2f9a80-6da6-11eb-8ecc-f404f1172e60.png)

HELP ME!!!
I don't know why the error!

**Comment by MasterKale** [comment]
Can you share the value of `body` that gets passed into `verifyAttestationResponse()` as `credential`? That error is thrown here:

https://github.com/MasterKale/SimpleWebAuthn/blob/master/packages/server/src/attestation/verifyAttestationResponse.ts#L67

This method assumes you're using **@simplewebauthn/browser**'s `startAttestation()` method to perform attestation on the client. As part of encoding the authenticator's response to a JSON-friendly format, the `rawId` buffer is encoded to base64url within `startAttestation()`. Since `id` is the base64url-encoded value of `rawId`, they should match.

If you're not using **@simplewebauthn/browser** in your project you should be able to still use `verifyAttestationResponse()` if you make sure `body.rawId` is base64url-encoded to match `body.id` before passing it in.

**Reply by nomad-pixel** [reply]
THANK you very much!!! I caught mistake!!!
